src/SmartTransactionsController.ts

```diff
@@ -566,36 +670,64 @@ export default class SmartTransactionsController extends BaseController<
       }
       const { nonceDetails } = nonceLock;
 
-      this.updateSmartTransaction({
-        chainId,
-        nonceDetails,
-        preTxBalance,
-        status: SmartTransactionStatuses.PENDING,
-        time,
-        txParams,
-        uuid: data.uuid,
-        cancellable: true,
-      });
+      this.#updateSmartTransaction(
+        {
+          nonceDetails,
+          preTxBalance,
+          status: SmartTransactionStatuses.PENDING,
+          time,
+          txParams,
+          uuid: data.uuid,
+          cancellable: true,
+        },
+        { chainId, ethQuery },
+      );
     } finally {
       nonceLock.releaseLock();
     }
 
     return data;
   }
 
+  getChainId({
+    networkClientId,
+  }: { networkClientId?: NetworkClientId } = {}): Hex {
+    return networkClientId
+      ? this.getNetworkClientById(networkClientId).configuration.chainId
+      : this.config.chainId;
+  }
+
+  getEthQuery({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  }): EthQuery {
+    return networkClientId
+      ? // @ts-expect-error TODO: Provider type alignment
+        new EthQuery(this.getNetworkClientById(networkClientId).provider)
+      : this.ethQuery;
+  }
+
   // TODO: This should return if the cancellation was on chain or not (for nonce management)
   // After this successful call client must update nonce representative
   // in transaction controller external transactions list
-  async cancelSmartTransaction(uuid: string): Promise<void> {
-    const { chainId } = this.config;
+  async cancelSmartTransaction(
+    uuid: string,
+    {
+      networkClientId,
+    }: {
+      networkClientId?: NetworkClientId;
+    } = {},
+  ): Promise<void> {
+    const chainId = this.getChainId({ networkClientId });
     await this.fetch(getAPIRequestURL(APIType.CANCEL, chainId), {
       method: 'POST',
       body: JSON.stringify({ uuid }),
     });
   }
 
-  async fetchLiveness(): Promise<boolean> {
-    const { chainId } = this.config;
+  async fetchLiveness(networkClientId?: NetworkClientId): Promise<boolean> {
```

Comment:
Do we want to put this in an options object too?

Reply:
Done here: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/94e2fac1cfdfedacd31b038a72815abb013dc19c